### PR TITLE
BREAKING CHANGE: create flight dataset(s)

### PR DESCRIPTION
**Pull request by yoctoyotta1024**
*** PART 3/3 of making a dataset from many .pds files: ***

Overview: new script for creating xarray dataset(s) from either :
a) raw .pds files,
b) .json files for serialised geometries of .pds files,
c) a .json file for a serialised flight geometry

- directly converts data that should be in 1 dataset but possibly spread over many .pds files into a single xarray dataset

new features besides this:
- a seperate dataset is created for each PPAR configuration that has data after it
-  decoders for creating a dataset use the PPAR configuration to decode data
- PPAR values are also written to final dataset

### CODE REVIEW

File: hamp_radar/flight_dataset.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+
+def postprocess_iq(ds):
+    # TODO(ALL): move to new file
+    return ds.pipe(decode_time)
+
+
+def read_datasetblock(datablock: DatasetGeometry, ppar: Optional[xr.Dataset] = None):
+    data = np.memmap(datablock.filename, mode="r")
+    raw_arrays = list(extract_raw_arrays(data, datablock.mainblock))
```
Comment (d70-t) [inline hamp_radar/flight_dataset.py:20]: This `list` seems unnecessary.

File: hamp_radar/serde_pds_files.py [hunk 2]
```diff
@@ -14,10 +14,11 @@ def scan_pdsfile(filename: Path):
 
 
 def serialize_write_data(writefile: Path, data):
+    import orjson
```
Comment (d70-t) [inline hamp_radar/serde_pds_files.py:17]: This seems like we need a new dependency in pyproject.toml

File: hamp_radar/decoders.py [hunk 3]
```diff
@@ -187,20 +175,34 @@ def _decode(rawdata):
...
 
 
-def decode_iq(ppar: Optional[xr.Dataset]):
-    nfft = 256
-    if ppar is not None:
-        nfft = ppar.sft.values  # assumed to be xr.Dataset
+def decode_fftd(rawdata, ppar: Optional[xr.Dataset] = None):
+    if ppar is None:
+        raise ValueError("need PPAR to decode FFTD")
+    # TODO raise eror is osc != [thing]
```
Comment (d70-t) [inline hamp_radar/decoders.py:181]: Would you implement this todo within this PR?

File: hamp_radar/serde_pds_files.py (hunk 2 above)
Comment (yoctoyotta1024) [reply hamp_radar/serde_pds_files.py:17]: yes, first we need a pyproject.toml... I'lll make serperate PR (commming soon)

File: hamp_radar/decoders.py (hunk 3 above)
Comment (yoctoyotta1024) [reply hamp_radar/decoders.py]: consider it done ;)

**Comment by yoctoyotta1024** [comment]
what d'you think of this PR now? @d70-t , ready to merge???

File: hamp_radar/decoders.py (hunk 3 above)
Comment (yoctoyotta1024) [reply hamp_radar/decoders.py]: (also doing radar tag check on decoders too)

File: hamp_radar/flight_dataset.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+
+def postprocess_iq(ds):
+    # TODO(ALL): move to new file
+    return ds.pipe(decode_time)
+
+
+def read_datasetblock(
+    datablock: DatasetGeometry,
+    radar_tag: Optional[str] = None,
```
Comment (d70-t) [inline hamp_radar/flight_dataset.py:21]: The `radar_tag` should already be part of the `mainblock` so we should not pass it in again a second time.

Comment (yoctoyotta1024) [reply hamp_radar/flight_dataset.py:21]: yes that's true but I thought it was clearer like this, although yes it's more error prone, will change...

Comment (yoctoyotta1024) [reply hamp_radar/flight_dataset.py]: better now?

**Review by d70-t — APPROVED** [review]
Awesome 🎉

**Pull request merged**